libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/NetworkEntitlement.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+import org.elasticsearch.core.Strings;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import static java.util.Map.entry;
+
+/**
+ * Describes a network entitlement (sockets) with actions.
+ */
+public class NetworkEntitlement implements Entitlement {
+
+    public static final int LISTEN_ACTION = 0x1;
+    public static final int CONNECT_ACTION = 0x2;
+    public static final int ACCEPT_ACTION = 0x4;
+
+    static final String LISTEN = "listen";
+    static final String CONNECT = "connect";
+    static final String ACCEPT = "accept";
+
+    private static final Map<String, Integer> ACTION_MAP = Map.ofEntries(
+        entry(LISTEN, LISTEN_ACTION),
+        entry(CONNECT, CONNECT_ACTION),
+        entry(ACCEPT, ACCEPT_ACTION)
+    );
+
+    private final int actions;
```

Comment:
Is the motivation here efficiency? `EnumSet` also packs each entry into a single bit, and is optimized by the jit, so it's the same efficiency, but much less code.

I had pictured this entitlement would look like this:

```
public record NetworkEntitlement(Set<NetworkAction> allowedActions) implements Entitlement{}
```

If you want this class to be responsible for parsing action strings, that could happen in a factory method rather than the constructor itself.

Reply:
Efficiency and consistency: FilePermission is done the same way.

Reply:
This seems the only "blocking" comment, so I'd like @rjernst  and @jdconrad  to chime in.

Reply:
Yes there would still be an object, but I'm sure the overhead would be unmeasurably small.